### ISSUE TITLE
maiko: Correct FPTEST definitions

### ISSUE THREAD

**Issue by waywardmonkeys** [Interlisp/medley#79]
With my current set of changes and one pending PR, for the cases where we aren't using `FLTINT`, we have these checks:

```
#else

/*  fpstatus_ is a FORTRAN library routine (in libc) which
    can be called to determine floating point status results.
    Documented in the Sun manual, "Floating Point Programmer's Guide",
    (Rev. A 19-Sep-86), pg. 34, it does *not* exist in libc for the
    SPARC.

    For sparc, should also check for isnan?  Don't know what isnormal
    & issubnormal do (these are sunos4.0 only)
 */
#if defined(OS5)
#define FPCLEAR
#define FPTEST(result) (!finite(result))

#elif defined(sparc) || defined(I386)
#define FPCLEAR
#define FPTEST(result) (isinf(result) || isnan(result))

#elif defined(LINUX)
#define FPCLEAR
#define FPTEST(result) ((!finite(result)) || isnan(result))

#elif defined(MACOSX) || defined(FREEBSD)
#include <math.h>
#define FPCLEAR
#define FPTEST(result) (!isfinite(result))

#elif defined(DOS)
#define FPCLEAR
#define FPTEST(result) (_getrealerror() & ( I87_ZERO_DIVIDE | I87_OVERFLOW | I87_UNDERFLOW))

#else
static int constant0 = 0;
unsigned int fpstatus_();

#define FPCLEAR         fpstatus_(&constant0);
#define FPTEST(result) (fpstatus_(&constant0) & 0xF0)

#endif
#endif /* FLTINT */
```

Back in historical times, the functions per platform were different indeed, but these days, they've been standardized and I think we can just do something much simpler here. But that then leaves the question of what should be checked!

If the macOS / FreeBSD code path is correct, then we can just be checking https://pubs.opengroup.org/onlinepubs/9699919799/functions/isfinite.html which will let us know if the value is not-infinite and not-nan.

But the Linux code is using the BSD functions `finite` and `isnan`, but according to https://linux.die.net/man/3/finite, `finite` should be sufficient.

I think we can throw away the `fpstatus_` usage which is from a SunOS fortran library from the late 1980s.

If we agree on the above, then the whole above block can become:

```
#elif defined(DOS)
#define FPCLEAR
#define FPTEST(result) (_getrealerror() & ( I87_ZERO_DIVIDE | I87_OVERFLOW | I87_UNDERFLOW))
#else
#include <math.h>
#define FPCLEAR
#define FPTEST(result) (!isfinite(result))
#endif /* FLTINT */
```

Does that match the expected semantics for Interlisp-D correctly?

**Comment by nbriggs** [comment]
Unfortunately, <code>isfinite()</code> is not available on Solaris with either "cc: Studio 12.6 Sun C 5.15 SunOS_sparc 2017/05/30" or "gcc (GCC) 7.3.0 [2017]" if you're compiling in c89/gnu89 compliance mode, which we are.

**Comment by waywardmonkeys** [comment]
This was improved in interlisp/maiko#127 and interlisp/maiko#82 as well as interlisp/maiko#63.

Going to mark this as closed for now.

The `OS5` code path must remain while we still build as C89.
